### PR TITLE
feat: quickFix for enum, const, property

### PR DESCRIPTION
**Pull request by p-spacek**
### What does this PR do?
add quick fix code-action for enums, consts, properties

Screenshot 2023-07-10 at 13 56 00

Screenshot 2023-07-10 at 13 55 16

### What issues does this PR fix or reference?
no ref

### Is it tested? How?
tests

### CODE REVIEW

File: src/languageservice/parser/jsonParser07.ts [hunk 1]
```diff
@@ -1385,6 +1387,7 @@ function validate(
                 length: propertyNode.keyNode.length,
               },
               severity: DiagnosticSeverity.Warning,
+              code: ErrorCode.PropertyExpected,
```
Comment (p-spacek) [inline src/languageservice/parser/jsonParser07.ts:1390]: not sure if it wouldn't be better to unify `data.properties` to `data.values` for the incorrect property

**Comment by p-spacek** [comment]
Hello @msivasubramaniaan, can I ask you for a review, please?

File: src/languageservice/services/yamlCodeActions.ts [hunk 2]
```diff
@@ -312,9 +316,52 @@ export class YamlCodeActions {
...
+    if (typeof diagnostic.data !== 'object') {
+      return;
+    }
+    if (
+      diagnostic.code === ErrorCode.EnumValueMismatch &&
+      'values' in diagnostic.data &&
+      Array.isArray((diagnostic.data as YamlDiagnosticData).values)
+    ) {
+      return (diagnostic.data as YamlDiagnosticData).values;
+    } else if (
```
Comment (gorkem) [inline src/languageservice/services/yamlCodeActions.ts:334]: Is this else really needed?

File: src/languageservice/services/yamlCodeActions.ts [hunk 3]
```diff
@@ -312,9 +316,52 @@ export class YamlCodeActions {
...
+    const results: CodeAction[] = [];
+    for (const diagnostic of diagnostics) {
+      const values = this.getPossibleQuickFixValues(diagnostic);
+      if (!values?.length) {
+        continue;
+      }
+      for (const value of values) {
+        results.push(
+          CodeAction.create(
+            value,
```
Comment (gorkem) [inline src/languageservice/services/yamlCodeActions.ts:353]: This returns only the value which is not very descriptive of the operation. perhaps we can replace it with something like 
```suggestion
            `Use value "${value}"`,
```

**Review by gorkem — CHANGES_REQUESTED** [review]

**Review by gorkem — CHANGES_REQUESTED** [review]
Only a couple of minor comments. It also needs to be rebased.

**Review by gorkem — CHANGES_REQUESTED** [review]
Only a couple of minor comments. It also needs to be rebased.

File: src/languageservice/services/yamlCodeActions.ts (hunk 2 above)
Comment (p-spacek) [reply src/languageservice/services/yamlCodeActions.ts:334]: Do you mean quickFix for properties? or can you explain your point further?
 - I think that property quickFix can be useful
 - note that there are different property names in `diagnostic.data` object (`values` vs `properties`)

File: src/languageservice/services/yamlCodeActions.ts (hunk 3 above)
Comment (p-spacek) [reply src/languageservice/services/yamlCodeActions.ts:353]: I don't have preferences here, I can remember that I tried a few combinations but I ended up with only `{value}`.
I was inspired by the English spelling checker extension (Code Spell Checker extension)
image

small disadvantage of `Use value "${value}"` is that this PR adds quickfix also for properties of the object (not value), so we need to display the text based on the error type
 - but for sure it's not a problem
 
 so please make a decision

**Comment by p-spacek** [comment]
@gorkem any updates here, please?
check updates and answers
Can you please rerun the jobs? It failed because of unavailable  jsonSchemaStore

**Review by gorkem — APPROVED** [review]

**Comment by msivasubramaniaan** [comment]
@p-spacek Please upstream the branch

**Comment by p-spacek** [comment]
> @p-spacek Please upstream the branch

@msivasubramaniaan I merged the main into this PR, but there is some schema change that caused the test to fail:

      -  "source": "yaml-schema: Composer Package"
      +  "source": "yaml-schema: Package"

**Pull request merged**